test/test_topology.py

```diff
@@ -621,6 +623,26 @@ def test_topology_repr(self):
             "<ServerDescription ('c', 27017) server_type: Unknown,"
             " rtt: None>]>" % (t._topology_id,))
 
+    def test_unexpected_load_balancer(self):
+        # Note: This behavior should not be reachable in practice but we
+        # should handle it gracefully nonetheless. See PYTHON-2791.
+        # Load balancers are included in topology with a single seed.
+        t = create_mock_topology(seeds=['a'])
+        mock_lb_response = {'ok': 1, 'msg': 'isdbgrid',
+                            'serviceId': ObjectId(), 'maxWireVersion': 13}
+        got_ismaster(t, ('a', 27017), mock_lb_response)
+        sds = t.description.server_descriptions()
+        self.assertIn(('a', 27017), sds)
+        self.assertEqual(sds[('a', 27017)].server_type_name, 'LoadBalancer')
+        self.assertEqual(t.description.topology_type_name, 'Single')
+        self.assertTrue(t.description.has_writable_server())
+
+        # Load balancers are removed from a topology with multiple seeds.
```

Comment:
What if both `a` and `b` advertise a serviceId?

Reply:
They'd both get removed. No need to test it though since this is unreachable in practice. We don't need to test every combination IMO.

Reply:
My reading of the code tells me that if the serviceId popping fails, we'll be left with one LB to which we'll connect as a standalone but I take your point that this is unreachable - we can leave it at that.